src/odemis/acq/align/fastem.py

```diff
@@ -273,95 +273,74 @@ def run(self):
 
         elif calibration == Calibrations.IMAGE_TRANSLATION_FINAL:
             d_offset = calibration.value.run(self._scanner, self._multibeam, self._descanner,
                                              self._detector, self._dataflow, self._ccd)
             self.asm_config["descanner"]["scanOffset"] = d_offset
-            configure_asm(self._multibeam, self._descanner, self._detector, self._dataflow, self.asm_config,
-                          upload=False)
 
         elif calibration == Calibrations.DARK_OFFSET:
             dark_offset = calibration.value.run(self._scanner, self._multibeam, self._descanner,
                                                 self._detector, self._dataflow)
             self.asm_config["mppc"]["cellDarkOffset"] = dark_offset
-            configure_asm(self._multibeam, self._descanner, self._detector, self._dataflow, self.asm_config,
-                          upload=False)
 
         elif calibration == Calibrations.DIGITAL_GAIN:
             digital_gain = calibration.value.run(self._scanner, self._multibeam, self._descanner,
                                                  self._detector, self._dataflow)
             self.asm_config["mppc"]["cellDigitalGain"] = digital_gain
-            configure_asm(self._multibeam, self._descanner, self._detector, self._dataflow, self.asm_config,
-                          upload=False)
 
         elif calibration == Calibrations.SCAN_ROTATION_FINAL:
             calibration.value.run(self._scanner, self._multibeam, self._descanner, self._detector, self._dataflow)
-            configure_asm(self._multibeam, self._descanner, self._detector, self._dataflow, self.asm_config,
-                          upload=False)
 
         elif calibration == Calibrations.SCAN_AMPLITUDE_FINAL:
             s_offset, s_amplitude = calibration.value.run(self._scanner, self._multibeam, self._descanner,
                                                           self._detector, self._dataflow)
             self.asm_config["multibeam"]["scanOffset"] = s_offset
             self.asm_config["multibeam"]["scanAmplitude"] = s_amplitude
-            configure_asm(self._multibeam, self._descanner, self._detector, self._dataflow, self.asm_config,
-                          upload=False)
 
         elif calibration == Calibrations.CELL_TRANSLATION:
             translation = calibration.value.run(self._scanner, self._multibeam, self._descanner, self._detector,
                                                 self._dataflow, plot_cell_translation=True)
             self.asm_config["mppc"]["cellTranslation"] = translation
-            configure_asm(self._multibeam, self._descanner, self._detector, self._dataflow, self.asm_config,
-                          upload=False)
+
+        else:
+            raise ValueError(f"Unknown calibration {calibration.name}.")
+
+        configure_asm(self._multibeam, self._descanner, self._detector, self._dataflow, self.asm_config, upload=False)
```

Comment:
BTW, why do you reconfigure the ASM after every calibration? Is it just in case we would need to run an acquisition afterwards? If so, shouldn't the acquisition do that step? Or is it to store some of the new calibration values?

Reply:
Good question. We needed to do so to set the new values on the VAs, which the next calibration was relying on. However, we also changed our implementation to make use of special MD that stores nowadays the values that should be used during the calibration to not have this dependency anymore. However, this line still makes sure that the calibrated values are set on the respective VAs and everything that is currently set on the VAs will be also used for the acquisition. So I think it needs to stay. @tepals what do you think?

I will add a comment to clarify why it is needed.

Reply:
I think it is indeed in case we need to run an acquisition afterwards. I'm not completely sure anymore why we left it like this and not move it into the acquisition code.

Reply:
If you're sure that each calibration is independent of the current VA value, then it should be safe.
In the code I see that also after running all the calibrations, `configure_asm()` is called one last time. So now it should be enough.

If you wanted to actually apply the settings just before the acquisition, then indeed this asm_config dict would need to obtained in some way. If I understand correctly, the "good" values are now stored as metadata. So it should be possible to have a similar function as `get_config_asm()`, but which would read the values from the metadata, instead of the VAs.

Reply:
I am a bit concerned with updating it now, especially since we do not have that much time anymore.

Reply:
I'd suggest to then leave it at the end of the function, and add a TODO indicating that it's probably not useful anymore (as each calib should be independent of the settings).

Reply:
ok will do this. I think the idea to obtain it from the MD sounds like a good idea.